src/Autosuggest.js

```diff
@@ -238,13 +262,17 @@ class Autosuggest extends Component {
             closeSuggestions('enter');
 
             if (focusedSuggestion !== null) {
+              const newValue = getSuggestionValue(focusedSuggestion);
+
               onSuggestionSelected(event, {
                 suggestion: focusedSuggestion,
-                suggestionValue: value,
+                suggestionValue: newValue,
                 sectionIndex: focusedSectionIndex,
                 method: 'enter'
               });
-              this.maybeCallOnSuggestionsUpdateRequested({ value, reason: 'enter' });
+
+              this.maybeCallOnChange(event, newValue, 'enter');
```

Comment:
I'm going to leave this as `user pressed Enter` as it will happen any time enter is pressed and the values are different. Using the suggested language here neglects to mention the case where you press enter when the first suggestion is auto-selected, and would require updating for any future cases with an increasingly long explanation.